sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -99,6 +104,56 @@ private[hive] object HiveShim {
     }
   }
 
+  def toMetastoreFilter(
+      predicates: Seq[Expression],
+      partitionKeys: List[FieldSchema],
+      hiveMetastoreVersion: String): Option[String] = {
+
+    // Binary comparison has been supported in getPartitionsByFilter() since Hive 0.13.
+    // So if Hive matastore version is older than 0.13, predicates cannot be pushed down.
+    // See HIVE-4888.
+    val versionPattern = "([\\d]+\\.[\\d]+).*".r
+    hiveMetastoreVersion match {
```

Comment:
Instead of checking the version here, I'd see if we can't move this into the [per version HiveShim](https://github.com/apache/spark/blob/master/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala) and just override it in the specific version classes, just so that we can be consistent with all of the other version specific code.